tracer/test/Datadog.Trace.Tests/DataStreamsMonitoring/DataStreamsWriterTests.cs

```diff
@@ -21,53 +23,87 @@ namespace Datadog.Trace.Tests.DataStreamsMonitoring;
 public class DataStreamsWriterTests
 {
     private const int BucketDurationMs = 1_000; // 1 second
+    private int _flushCount;
 
-    [Fact]
-    public async Task DoesNotWriteIfNoStats_Periodic()
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    [InlineData(null)]
+    public async Task DoesNotWriteIfNoStats_Periodic(bool? isSupported)
     {
         var bucketDurationMs = 100; // 100 ms
         var api = new StubApi();
-        var writer = CreateWriter(api, bucketDurationMs * 1_000_000);
+        var writer = CreateWriter(api, out var discovery, bucketDurationMs);
+        writer.FlushComplete += (_, _) => Interlocked.Increment(ref _flushCount);
+        TriggerSupportUpdate(discovery, isSupported);
 
-        await Task.Delay(bucketDurationMs * 10);
+        await WaitForFlushCount(bucketDurationMs * 10);
 
         api.Sent.Should().BeEmpty();
     }
 
-    [Fact]
-    public async Task DoesNotWriteIfNoStats_OnClose()
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    [InlineData(null)]
+    public async Task DoesNotWriteIfNoStats_OnClose(bool? isSupported)
     {
         var bucketDuration = 100_000_000; // 100 ms
```

Comment:
nitpicky (and outside of the PR)

```suggestion
        var bucketDuration = 100_000_000;
```